runtime/oti/j9javaaccessflags.h

```diff
@@ -163,14 +163,35 @@
 #define J9StaticFieldRefTypeShort 0x4
 #define J9StaticFieldRefTypeIntFloat 0x5
 #define J9StaticFieldRefTypeLongDouble 0x6
-#define J9StaticFieldRefTypeUnused_0x7 0x7
+#if defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES)
+#define J9StaticFieldIsNullRestricted 0x7
+#endif /* defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES) */
 #define J9StaticFieldRefVolatile 0x8
 #define J9StaticFieldRefPutResolved 0x10
 #define J9StaticFieldRefFinal 0x20
-#if defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES)
-#define J9StaticFieldIsNullRestricted 0x40
-#endif /* defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES) */
-#define J9StaticFieldRefUnused_0x80 0x80
+#if defined(J9VM_OPT_VALHALLA_STRICT_FIELDS)
+/**
+ * Since only two flags are available they are
+ * used as follows to represent three states
+ * for strict static fields:
+ * 00XXXXXX - field is not strict
+ * 01XXXXXX - strict field has been read. This implies it was
+ * written at least once as well otherwise an exception would have
+ * been thrown.
+ * 10XXXXXX - strict field has been written at least once.
+ * 11XXXXXX - strict field has not been read or written.
+ */
+#define J9StaticFieldRefStrictInit1 0x40
+#define J9StaticFieldRefStrictInit2 0x80
+#define J9_STATIC_FIELD_STRICT_INIT_IS_UNSET(classAndFlags) \
+    J9_ARE_ALL_BITS_SET(classAndFlags, J9StaticFieldRefStrictInit1 | J9StaticFieldRefStrictInit2)
+#define J9_STATIC_FIELD_STRICT_INIT_WAS_WRITTEN_AND_READ(classAndFlags) \
+    (J9_ARE_ANY_BITS_SET(classAndFlags, J9StaticFieldRefStrictInit1) \
+    && J9_ARE_NO_BITS_SET(classAndFlags, J9StaticFieldRefStrictInit2))
+#define J9_STATIC_FIELD_STRICT_INIT_WAS_WRITTEN(classAndFlags) \
+    (J9_ARE_NO_BITS_SET(classAndFlags, J9StaticFieldRefStrictInit1) \
+    && J9_ARE_ANY_BITS_SET(classAndFlags, J9StaticFieldRefStrictInit2))
```

Comment:
Rather than testing the two bits separately, suggest they be masked and compared in a single test, for example:
```c
#define J9StaticFieldRefStrictMask 0xC0
#define J9StaticFieldRefStrictRead 0x40
[...]
#define J9_STATIC_FIELD_STRICT_INIT_WAS_WRITTEN_AND_READ(classAndFlags) \
	(J9StaticFieldRefStrictRead == ((classAndFlags) & J9StaticFieldRefStrictMask))
```